### DISCUSSION TITLE
Are GeoCore Ver. 18.02.0 files available?

### DISCUSSION THREAD

**Discussion by mtbluedog** [geodesicsolutions-community/geocore-community#225] [Q&A]
I've got corrupt and/or lost files in a GeoCore 18.02.0 install; is 18.02.0 available for download anywhere?

**Comment by vicos59** [comment]
I might have them somewhere, I'll have to look.

Another thing to check is that often times the admin directory was renamed by the installer to something else, for security reasons. I think It the admin directory is specified in the config file, but its been a long time, so....

You have to be aware of this and take extra steps during an upgrade if this is the case.

**Reply by mtbluedog** [reply]
Would really appreciate the files or an archive of v18!
I'll take a look and see if I can tell if admin has been renamed.

**Reply by vicos59** [reply]
It's right here: https://github.com/geodesicsolutions-community/geocore-community/releases/tag/v0.0.0

**Reply by mtbluedog** [reply]
Thanks! "admin" is designated in the config file.

So many of the files in my original install are different, according to a folder to folder comparison; I have no idea what happened. The admin login says 18.02.0.

Can I simply copy all the fresh v18 files and overwrite everything, except for the geo_templates directory?

Is v18 PHP7 compatible? Or does it require PHP5.6 max?

**Comment by vicos59** [comment]
I'm not really following exactly where you are at and where you want to go. It sounds like maybe you inherited a non-functional v18.02 site (outside of just licensing issues)? If that's the case, I don't really understand what is and isn't working at this point in time.

Were it me, I would just install the license patch, then do the upgrades to v20. The upgrade will install all of the proper files. Of course, this all assumes that your database is not corrupted.

Whatever you do, FOLLOW THE INSTRUCTIONS VERBATIM when doing either an upgrade or fresh install. Otherwise, you are headed for trouble. The instructions are on the GeoCore website.

Also, be sure to have backups in case something goes horribly wrong and you need to start over. If you're totally lost, you can always hire @blufyremedia to fix your site for you.

**Comment by mtbluedog** [comment]
Thanks, I had a non-functional v18.02 site (outside of licensing issues), but I copied over fresh files (except templates) from the v18 here `[releases/tag/v0.0.0](https://github.com/geodesicsolutions-community/geocore-community/releases/tag/v0.0.0)` and it mostly works now, and I can now log into admin, so that's fixed. 

I'm still getting one missing file error. The file `classes/php5_classes/ListingDisplay.class.php` calls the template file `mapping_directions_link_google.tpl`, but that template file doesn't exist in v18 or v20. Any ideas where I can get that file? Is it even used anymore? @blufyremedia 

I don't want to try an upgrade to v20 until v18 works.

**Reply by wrangster** [reply]
I had the same problem, and have not solved it. I think Jonyo said that it has to do with some modules needs, or are in the middle of upgrading. The file you ask for does not exist and never have existed.

**Reply by vicos59** [reply]
>> mapping_directions_link_google.tpl

Sounds like that is part of the Google Maps Addon/Plugin. Try going to your Addons section in admin admin see if that is installed/enabled. If it is, maybe toggle it off or vice versa.

**Reply by mtbluedog** [reply]
Good idea; add-on turned off, but still have the error.
Like @wrangster said, this file has never existed.
Can I remove the function from ListingDisplay.class.php that calls the template?
Could I add a blank template file called mapping_directions_link_google.tpl that has nothing in it?

**Reply by wrangster** [reply]
I think it's enough to delete the map addon tag in listing page, at least that's what I did.

**Reply by vicos59** [reply]
My last Hail Mary is to:

Admin-->Listing Setup-->Fields to Use--> Unenable "Mapping Location"

**Reply by vicos59** [reply]
>> I think it's enough to delete the map addon tag in listing page, at least that's what I did.

That sounds like the simple solution.

**Reply by mtbluedog** [reply]
"delete the map addon tag in listing page"
In Admin? Or in the page template?
"Admin-->Listing Setup-->Fields to Use--> Unenable "Mapping Location""
Did that, no luck.

Also, site cache is off.

**Reply by vicos59** [reply]
Can you post the entire error message? 

I don't have this issue on my system, so its hard to understand why others do.

**Reply by wrangster** [reply]
I meant in the page template. The one that looks like this:
{addon author='geo_addons' addon='google_maps' tag='listing_map' assign='map'}

**Reply by vicos59** [reply]
![image](https://github.com/geodesicsolutions-community/geocore-community/assets/6634041/5fcf6420-c087-4194-84f4-9cbec614c30b)

**Reply by mtbluedog** [reply]
Thanks, but this is still a mystery.
I searched all themes and deleted all instances of `addon='google_maps'` and deactivated the add-on, still get this error when going to an individual listing on the front end: 
![error](https://github.com/geodesicsolutions-community/geocore-community/assets/23107911/c10a4f3e-0a74-46d1-8313-ddf44a97008e)
Can I create an empty .tpl file and what folder would it go in?
The file classes/php5_classes/ListingDisplay.class.php calls the template file mapping_directions_link_google.tpl Can I edit that function out of ListingDisplay.class.php ?

**Reply by wrangster** [reply]
I tried to rename another google maps file, but it still gave the same error. It's a while since I did this now, but I think you maby have to delete more than only the addon tag. Remove everything that has to do with google maps, and then I think you are ok. Dont be so afraid to try things, you can allways just drop in a fresh copy of the file if you screw something up. :)

**Reply by wrangster** [reply]
This script looks like it asks for some google stuff, maby it is the problem. On line 178 in the default listing_classified.tpl. 

<script>
							jQuery('#tab_map_link').on('shown.bs.tab', function() {
								var center = addon_google_maps.mapHook.getCenter();
								google.maps.event.trigger(addon_google_maps.mapHook, 'resize');
								addon_google_maps.mapHook.setCenter(center);
							});
						</script>

**Reply by mtbluedog** [reply]
Thanks, that's just jQuery, it won't cause a PHP error, only throw a console error.

**Comment by vicos59** [comment]
>>Can I create an empty .tpl file and what folder would it go in? I would just use the Template editor in admin and create it there. Go to the 'system' folder of the template set you are using. I don't see a function to create a new file, so maybe copy an existing one and rename it and then empty it out.

![Screenshot 2024-01-16 135000](https://github.com/geodesicsolutions-community/geocore-community/assets/6634041/0b455a73-4792-473a-8c0a-3ea904df42f1)

>> The file classes/php5_classes/ListingDisplay.class.php calls the template file mapping_directions_link_google.tpl Can I edit that function out of ListingDisplay.class.php ?

Just give it a try. Just make note because future updates may overwrite this file and eradicate your changes.

This bug must have been fixed in v20, because I do not experience it.

**Reply by mtbluedog** [reply]
Thanks, well, we're getting somewhere :)

I had to add the blank file mapping_directions_link_google.tpl in the default theme, not the custom theme, where it didn't work. And I had to add a header to it, i.e. {* 7.0.3-314-g84cfb37 *} to make it work.

So now the front end loads without that error, but none of the Javascripts, styesheets or images load. They 404 because the URL is absolute, i.e. test.example.com/category/... geo_templates/default/external/js/bootstrap.min.js, not relative, like they should be, i.e. test.example.com/geo_templates/default/external/js/bootstrap.min.js

So that's strange. I am running this site on a subdomain, is that the problem?
